test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -158,22 +159,6 @@ def test_proxy_conn_fail_from_dns(
                 == urllib3.exceptions.NameResolutionError
             )
 
-    @pytest.mark.parametrize("target_scheme", ["http", "https"])
```

Comment:
Because timeouts can come in multiple flavors (socket.timeout, ssl.SSLError with a timeout errno) I think it makes sense to wrap these errors, even within a `ProxyError`. Also you wouldn't know whether the timeout was on read or connect.